### PR TITLE
In continuation to PR1243 to solve issue #1240

### PR DESCRIPTION
**Pull request by Saanidhyavats**
## Proposed changes
Solves #1240 
Please include a description of the problem or feature this PR is addressing. If there is a corresponding issue, include the issue #.

## Checklist

Put an `x` in the boxes that apply.

- [x ] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x ] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x ] I have added tests that prove my fix is effective or that my feature works
- [x ] I have updated the necessary documentation (if needed)

### CODE REVIEW

**Comment by Saanidhyavats** [comment]
I have made changes to two functions in the codebase. Before proceeding with further modifications, I would appreciate it if you could review these changes to ensure they are correct.

File: python/mlx/utils.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def tree_flatten(tree, prefix="", is_leaf=None):
     return [(prefix[1:], tree)]
 
 
-def tree_unflatten(tree):
+def tree_unflatten(tree: Any) -> Any:
```
Comment (awni) [inline python/mlx/utils.py:158]: The type here for tree does not match the type in the doc string. I would use the type in the docstring which is more specific.

Comment (Saanidhyavats) [reply python/mlx/utils.py]: Thank you for letting me know, I just added the change

File: python/mlx/utils.py [hunk 2]
```diff
@@ -111,7 +111,7 @@ def tree_map_with_path(
         return fn(path, tree, *rest)
 
 
-def tree_flatten(tree, prefix="", is_leaf=None):
+def tree_flatten(tree: Any, prefix: str = "", is_leaf: Callable = None) -> Any:
```
Comment (awni) [inline python/mlx/utils.py:114]:
```suggestion
def tree_flatten(tree: Any, prefix: str = "", is_leaf: Optional[Callable] = None) -> Any:
```

File: python/mlx/utils.py [hunk 3]
```diff
@@ -111,7 +111,7 @@ def tree_map_with_path(
         return fn(path, tree, *rest)
 
 
-def tree_flatten(tree, prefix="", is_leaf=None):
+def tree_flatten(tree: Any, prefix: str = "", is_leaf: Optional[Callable] = None) -> Any:
```
Comment (awni) [inline python/mlx/utils.py:114]:
```suggestion
def tree_flatten(
    tree: Any, prefix: str = "", is_leaf: Optional[Callable] = None
) -> Any:
```

**Review by awni — APPROVED** [review]
Thanks!

**Comment by Saanidhyavats** [comment]
I will be adding more changes

**Comment by awni** [comment]
Great please to start a new or for that

**Pull request merged**